src/main/java/pl/squirrel/classycle/ClassyclePlugin.java

```diff
@@ -1,85 +1,65 @@
 package pl.squirrel.classycle;
 
-import classycle.ant.DependencyCheckingTask;
-import org.apache.tools.ant.types.FileSet;
-import org.gradle.api.Action;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
-import org.gradle.api.logging.Logger;
 import org.gradle.api.plugins.JavaPluginConvention;
 import org.gradle.api.reporting.ReportingExtension;
 import org.gradle.api.tasks.SourceSet;
-import org.gradle.internal.UncheckedException;
 
 import java.io.File;
-import java.net.URI;
-import java.net.URISyntaxException;
 import java.util.SortedMap;
 
 public class ClassyclePlugin implements Plugin<Project> {
 
+    public static final String TASK_GROUP_NAME = "Verification";
+
+    private ClassyclePluginConfiguration configuration;
+
     @Override
     public void apply(final Project project) {
-        final Logger log = project.getLogger();
+        loadConfiguration(project);
+
         final JavaPluginConvention javaPlugin = project.getConvention().getPlugin(JavaPluginConvention.class);
         final ReportingExtension reporting = project.getExtensions().getByType(ReportingExtension.class);
         final SortedMap<String, SourceSet> sourceSets = javaPlugin.getSourceSets().getAsMap();
 
+        File projectDir = project.getProjectDir();
+        org.apache.tools.ant.Project antProject = project.getAnt().getAntProject();
+
         final Task classycleTask = project.task("classycle");
+        classycleTask.setGroup(TASK_GROUP_NAME);
+        classycleTask.setDescription("Run all classycle tasks");
         final Task checkTask = project.getTasks().getByName("check");
         for (String name : sourceSets.keySet()) {
```

Comment:
Actually, you may still want to have some quality checks for the test code as well. They may be different (that's why the definition file is classycle-main.txt, classycle-test.txt, etc.), but this can still be pretty useful.

Also, you may have other source sets there, not just "main" and "test".